Kafka broker ordered delivery (#3369)
knative/docs: docs/eventing/broker/kafka-broker.md

    kubectl rollout restart deployment -n knative-eventing kafka-broker-dispatcher
    ```

## Events Delivery order

> When dispatching events, Kafka Broker supports two different ordering guarantees:

Reviewer: ```suggestion
When dispatching events, Kafka broker supports two different ordering guarantees:
```

> * `unordered`: Unordered consumer is a non-blocking consumer that potentially deliver messages unordered, while preserving proper offset management. 

Reviewer 2: ```suggestion
* `unordered`: Unordered consumer is a non-blocking consumer that potentially delivers messages unordered, while preserving proper offset management. 
```

> * `ordered`: Ordered consumer is a per-partition blocking consumer that deliver messages in order.

Reviewer 2: ```suggestion
* `ordered`: Ordered consumer is a per-partition blocking consumer that delivers messages in order.
```

> * `ordered`: Ordered consumer is a per-partition blocking consumer that delivers messages in order.

Reviewer: we call these "ordering guarantees" above, but then "consumers" here. Can we choose a single term here for referring to this or explain the relationship better?

Author: I've added _consumer_ above to refer to the fact that we're talking about the kafka broker consumer delivery guarantees

`unordered` is the default ordering guarantee, while **`ordered` is considered unstable, use with caution**.

> You can configure the delivery order guarantee using the `kafka.eventing.knative.dev/delivery.order` annotation on the `Trigger` instance.

Reviewer: ```suggestion
You can configure the delivery order of events using the `kafka.eventing.knative.dev/delivery.order` annotation on the `Trigger` object.
```

For example:

```yaml
apiVersion: eventing.knative.dev/v1
kind: Trigger
metadata:
  name: my-service-trigger
  annotations:
     kafka.eventing.knative.dev/delivery.order: ordered
spec:
  broker: my-kafka-broker
  subscriber:
    ref:
      apiVersion: serving.knative.dev/v1
      kind: Service
      name: my-service

> ```

Reviewer: I'd move this to the start and explain what ordered and unordered mean after it's shown where they are configured to provide context

Author: How about now?

### Additional information

> - To report bugs or add feature requests, open an issue in the [eventing-kafka-broker repository](https://github.com/knative-sandbox/eventing-kafka-broker).

Reviewer: ```suggestion
- To report a bug or request a feature, open an issue in the [`eventing-kafka-broker` repository](https://github.com/knative-sandbox/eventing-kafka-broker).
```
